### PR TITLE
AO3-4592 Un-hide works affected by anon/unrevealed collection invitation bug

### PR DESCRIPTION
**Pull request by zz9pzza**
https://otwarchive.atlassian.net/browse/AO3-4592

### CODE REVIEW

**Comment by zz9pzza** [comment]
I do want someone who knows about challenges to look at this.

**Comment by sarken** [comment]
@shalott or @scottsds, you know the collection invitation bug code best -- can you take a look?

**Comment by shalott** [comment]
I'm wiped out and a bit braindead but I'm pretty sure this will set works to anon/unrevealed even if they have rejected an invitation to a collection. You have to follow the model in collectible.rb/update_anon_unrevealed and only use approved collection items to set the status. 

I am not sure but if update_anon_unrevealed doesn't cause a 'changed?' flag on a work if the value has changed, you could just do something like:
```
Works.find_each do |w|
  w.update_anon_unrevealed
  w.save if w.changed?
end

```
but I am not sure that won't cause the w.changed? to always return true even if it wasn't changed, it would have to be tested in console first to make sure so we don't end up saving every single work object over. :P

**Comment by shalott** [comment]
Er, I mean, if update_anon_unrevealed doesn't cause a 'changed?' flag on a work if the value has NOT changed. :P

**Comment by sarken** [comment]
Just wanted to poke and see how this was coming! It would be really nice to fix the works that have been hidden for months /o\

**Comment by sarken** [comment]
I've checked it on the console and it seems like that will work fine -- it only saves the works that actually changed. Instead of doing `update_anon_unrevealed` on every work, do we want to just do it on works where `in_anon_collection` and/or `in_unrevealed_collection` is true?

**Comment by sarken** [comment]
Also! We should probably just put this in the file with the after tasks, since it's a one-time thing.

**Comment by sarken** [comment]
@zz9pzza https://github.com/sarken/otwarchive/tree/AO3-4592 contains the fix, but whenever I try to open a PR on your branch, it picks up 1000+ extra unwanted commits

File: lib/tasks/after_tasks.rake [hunk 1]
```diff
@@ -626,4 +638,4 @@ end # this is the end that you have to put new tasks above
 desc "Run all current migrate tasks"
 # task :After => ['After:convert_tag_sets', 'autocomplete:reload_tagset_data', 'skins:disable_all', 'skins:unapprove_all',
 # 'skins:load_site_skins', 'After:convert_existing_skins', 'skins:load_user_skins', 'After:remove_old_epubs']
-task :After => ['After:locale_setup']
+task :After => ['After:locale_setup']
```
[the remark this answers is not in the document]
Comment (redsummernight) [reply lib/tasks/after_tasks.rake]: We should follow Hound here: add the final newline, and also remove the newly added trailing whitespaces.

**Pull request merged**